component/src/main/java/org/wso2/extension/siddhi/execution/ml/classification/perceptron/util/PerceptronModel.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.extension.siddhi.execution.ml.classification.perceptron.util;
+
+import org.wso2.extension.siddhi.execution.ml.util.MathUtil;
+
+import java.io.Serializable;
+import java.util.Arrays;
+
+/**
+ * Represents a linear Perceptron Model
+ */
+public class PerceptronModel implements Serializable {
+    private static final long serialVersionUID = -5179648194841293764L;
+    private double[] weights;
+    private double bias = 0.0;
+    private double threshold = 0.5;
+    private double learningRate = 0.1;
+
+    public PerceptronModel() {
+    }
+
+    public PerceptronModel(PerceptronModel model) {
+        this.weights = model.weights;
+        this.bias = model.bias;
+        this.threshold = model.threshold;
+        this.learningRate = model.learningRate;
+    }
+
+    public double[] update(Boolean label, double[] features) {
```

Comment:
It's a good practice to have java docs for at least public methods.

Reply:
Fixing with this PR [#51](https://github.com/wso2-extensions/siddhi-execution-streamingml/pull/51)